inttest/calico/calico_test.go

```diff
@@ -58,6 +62,40 @@ func (s *CalicoSuite) TestK0sGetsUp() {
 
 	s.T().Log("waiting to see calico pods ready")
 	s.NoError(common.WaitForDaemonSet(s.Context(), kc, "calico-node"), "calico did not start")
+
+	targetPod, err := kc.CoreV1().Pods("default").Create(s.Context(), &corev1.Pod{
+		TypeMeta:   metav1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "hostname-pod"},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{{Name: "hostname", Image: "registry.k8s.io/serve_hostname"}},
+			NodeSelector: map[string]string{
+				"kubernetes.io/hostname": "worker0",
+			},
+		},
+	}, metav1.CreateOptions{})
+	s.Require().NoError(err)
+	s.NoError(common.WaitForPod(s.Context(), kc, "hostname-pod", "default"), "hostname pod did not start")
+
+	sourcePod, err := kc.CoreV1().Pods("default").Create(s.Context(), &corev1.Pod{
+		TypeMeta:   metav1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "busybox"},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Name:    "busybox",
+				Image:   "gcr.io/google-containers/busybox",
+				Command: []string{"sleep", "600000"},
+			}},
+			NodeSelector: map[string]string{
+				"kubernetes.io/hostname": "worker1",
+			},
+		},
+	}, metav1.CreateOptions{})
+	s.Require().NoError(err)
+	s.NoError(common.WaitForPod(s.Context(), kc, "busybox", "default"), "busybox pod did not start")
+
+	out, err := common.PodExecCmdOutput(kc, restConfig, sourcePod.Name, sourcePod.Namespace, fmt.Sprintf("/usr/bin/wget -qO- %s:9376", targetPod.Status.PodIP))
```

Comment:
This might be bit flaky as such. I think we should wait till konnectivity has its tunnels setup before we do pod exec. One "trick" done in some other smokes is to wait to see some pod logs as that exercises the konnecivity tunnel.

Reply:
I don't think it will be flaky since first we wait for the calico daemonset to be ready and then spin up two pods, but yeah, I added `WaitForLogs`.

Reply:
Well, the pod exec's etc. being flaky has nothing to do with if calico is ready or not. exec goes via konnectivity tunnel and if that's not fully working the execs will fail. But waiting to get logs has turned to be a good signal that the tunnels are up properly.